cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/WorkflowStatsSlaHttpHandler.java

```diff
@@ -0,0 +1,248 @@
+import co.cask.cdap.internal.app.store.WorkflowDataset;
+import co.cask.cdap.proto.Id;
+import co.cask.cdap.proto.ProgramType;
+import co.cask.http.AbstractHttpHandler;
+import co.cask.http.HttpResponder;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import org.jboss.netty.handler.codec.http.HttpRequest;
+import org.jboss.netty.handler.codec.http.HttpResponseStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+
+/**
+ * Workflow Statistics Handler
+ */
+@Singleton
+@Path(Constants.Gateway.API_VERSION_3 + "/namespaces/{namespace-id}")
+public class WorkflowStatsSlaHttpHandler extends AbstractHttpHandler {
+
+  private static final Logger LOG = LoggerFactory.getLogger(WorkflowStatsSlaHttpHandler.class);
+  protected final Store store;
+  private final MetricStore metricStore;
+
+  @Inject
+  public WorkflowStatsSlaHttpHandler(MetricStore metricStore, Store store) {
+    this.metricStore = metricStore;
+    this.store = store;
+  }
+
+  @GET
+  @Path("apps/{app-id}/workflows/{workflow-id}/statistics")
+  public void workflowStats(HttpRequest request, HttpResponder responder,
+                            @PathParam("namespace-id") String namespaceId,
+                            @PathParam("app-id") String appId,
+                            @PathParam("workflow-id") String workflowId,
+                            @QueryParam("start") String start,
+                            @QueryParam("end") String end,
+                            @QueryParam("percentile") List<Double> percentiles) throws Exception {
+
+    long startTime = TimeMathParser.parseTimeInSeconds(start);
+    long endTime = TimeMathParser.parseTimeInSeconds(end);
+
+    if (endTime < startTime || startTime < 0 || endTime < 0) {
+      throw new BadRequestException("Wrong start or end time.");
+    }
+
+    for (double i : percentiles) {
+      if (i < 0 || i > 100) {
+        throw new BadRequestException("Percentile values have to be greater than 0 and less than 100");
+      }
+    }
+
+    List<WorkflowDataset.WorkflowRunRecord> workflowRunRecords;
+
+    Id.Program programId = Id.Program.from(namespaceId, appId, ProgramType.WORKFLOW, workflowId);
+    try {
+      try {
+        workflowRunRecords = store.getWorkflowRuns(programId, startTime, endTime);
+
+      } catch (IllegalArgumentException exception) {
+        responder.sendJson(HttpResponseStatus.BAD_REQUEST, "Bad request");
+        return;
+      }
+    } catch (SecurityException exception) {
+      responder.sendJson(HttpResponseStatus.BAD_REQUEST, "You're not allowed to view this");
```

Comment:
Instead of using responder.sendJson(..., String), use responder.sendString(...).
My concern is that the string you respond with might get Gson serialized additionally when using the sendJson method (it'd have extra quotes around it, like `"\""You're not allowed to view this\""`).